cishouseholds/validate_class.py

```diff
@@ -179,19 +292,43 @@ def between(error_message, column_name, range_set):
 
     # Non column wise functions
     @staticmethod
-    def duplicated(error_message, check_columns):
+    def duplicated(error_message: str, check_columns: List[str]):
+        """
+        Finds duplicated values by given column and can specify specific error message.
+        Parameters
+        ----------
+        error_message
+        check_columns
+        """
         window = Window.partitionBy(*check_columns)
         error_message = error_message.format(", ".join(check_columns))
         return F.when(F.sum(F.lit(1)).over(window) == 1, True).otherwise(False), error_message
 
     @staticmethod
-    def valid_vaccination(error_message, survey_response_type_column, check_columns):
-        return (F.col(survey_response_type_column) != "First Visit") | (
-            ~F.array_contains(F.array(*check_columns), None)
-        ), error_message
+    def valid_vaccination(error_message: str, survey_response_type_column: str, check_columns: List[str]):
+        """
+        works out valid vaccination by finding "First Visit" or None TODO
+        Parameters
+        ----------
+        error_message
+        survey_response_type_column
+        check_columns
+        """
+        return (
+            (F.col(survey_response_type_column) != "First Visit") | (~F.array_contains(F.array(*check_columns), None)),
+            error_message,
+        )
 
     @staticmethod
     def check_all_null_given_condition(error_message: str, condition: Any, null_columns: List[str]):
+        """
+        TODO
```

Comment:
check all columns in list are null

Reply:
the check for null should probably be refactored to use the check for nulls in rows in expressions